packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -33,26 +34,76 @@ function getNodeProperty(node, path) {
 }
 
 /**
- * Get a normalized file path. If Embroider prefix is present, strip it out
+ * Function to parse defaults.project and only return info to be used by the plugin
+ * @param {object} project Ember defaults.project
+ * @returns {object} Contains project name and ember-addon path info
+ */
+function getProjectInfo(project) {
+  const parsedProjectInfo = {};
+
+  if (project) {
+    parsedProjectInfo.pkg = {
+      name: project.pkg.name,
+      'ember-addon': {
+        paths: project.pkg['ember-addon'].paths,
+      },
+    };
+  }
+
+  return parsedProjectInfo;
+}
+
+/**
+ * Strip out project name prefix segement(s) from file path
+ * @param {array} tokens File path segments
+ * @param {object} projectInfo Contains project name, ember-addon path info
+ * @returns {string} A joined file path, e.g. tests/acceptance/my-test.js
+ */
+function getParsedClassicFilepath(tokens, projectInfo) {
+  const projectNameTokens = projectInfo.pkg.name.split(path.sep);
+
+  tokens.splice(0, tokens.indexOf(projectNameTokens[0]) + projectNameTokens.length);
+
+  return tokens.join(path.sep);
+}
+
+/**
+ * Strip out Embroider prefix (embroider/nnnnnn) segements from file path
+ * @param {array} tokens File path segments
+ * @returns {string} A joined file path, e.g. tests/acceptance/my-test.js
+ */
+function getParsedEmbroiderFilepath(tokens) {
+  const RELATIVE_PATH_ROOT = 2;
+
+  tokens.splice(0, tokens.lastIndexOf(EMBROIDER) + RELATIVE_PATH_ROOT);
+
+  return tokens.join(path.sep);
+}
+
+/**
+ * Get a normalized file path, based on whether the app build is classic or with Embroider
  * @param {object} fileOpts Babel state.file.opts which include root and filename props
+ * @param {object} projectInfo Contains project name, ember-addon path info
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
-function getNormalizedFilePath(fileOpts) {
+function getNormalizedFilePath(fileOpts, projectInfo) {
   let { root, filename } = fileOpts;
   const tokens = filename.split(path.sep);
-  const EMBROIDER = 'embroider';
+  const isEmbroider = tokens.includes(EMBROIDER);
```

Comment:
How is it buggy? Can you elaborate on this?

That path you showed would never exist, since we have intimate knowledge of that build system and the paths it produces. You'd never have a nested reference to `embroider` inside a directory, and if you do, that's bonkers and you deserve to have your stuff break :)

Reply:
it's brittle because its coupling to something even more transient than a private API, its bound essentially to an internal constant string that the maintainer could change without any expectation that it'd break things. certainly we can say that should never happen, but that doesn't make it less brittle.

Reply:
It's a consistent part of the build system for embroider. I do acknowledge that a env var or something similar could be better, but in lieu of that this seems reasonable. Also, we've run this past those who are building embroider and this was the recommended approach.

If we feel really strongly about this, we can also pass a boolean value in to the plugin to indicate whether it's using embroider or not. It can easily introspect to determine whether the app is using embroider since the module where we'd pass the options into this babel plugin would be the same module where embroider APIs are being used. WDYT?

Reply:
that makes sense and thanks for the info. this is a special case in a way, and I'm happy to take the recommended approach.